SeaPublicWebsite/Controllers/AuthController.cs

```diff
@@ -45,6 +45,12 @@ public IActionResult Index_Post(AuthViewModel viewModel)
             HttpOnly = true
         });
 
-        return Redirect(viewModel.ReturnPath);
+        // links starting with // are protocol-relative URLs which can be used to redirect to other domains
+        // send to index if this is the case
+        var returnPath = viewModel.ReturnPath.StartsWith("//")
+            ? "/"
+            : viewModel.ReturnPath;
```

Comment:
I didn't realise we weren't also guarding against general https:// redirects.

Eg. see https://dev.find-energy-improvements-suitable-for-your-home.service.gov.uk/password?returnPath=https://google.com

Something like:
```suggestion
        var returnPath = viewModel.ReturnPath.StartsWith("//") || !viewModel.ReturnPath.StartsWith("/")
            ? "/"
            : viewModel.ReturnPath;
```

Reply:
ah good point. have accounted for this now